### PR TITLE
Improving the performance of TF models for aarch64.

### PR DESCRIPTION
**Pull request by renato-arantes**
The aim of this PR is to enhance the efficiency of TF models. The following chart showcases the progress achieved through our modifications by presenting a contrast between ACL and Eigen across different NN architectures and shapes.

<p align="center">

</p>

### CODE REVIEW

**Comment by nSircombe** [comment]
Hi @penpornk - some performance improvements to the AArch64 build with oneDNN, would you be able to take a look please?

File: tensorflow/core/common_runtime/mkl_layout_pass.cc [hunk 1]
```diff
@@ -394,7 +394,7 @@ class MklLayoutRewritePass : public GraphOptimizationPass {
          CopyAttrsAll, RewriteIfAtleastOneMklInput, GetRewriteCause()});
     rinfo_.push_back({csinfo_.avg_pool,
                       mkl_op_registry::GetMklOpName(csinfo_.avg_pool),
-                      CopyAttrsAll, AlwaysRewrite, GetRewriteCause()});
+                      CopyAttrsAll, CheckRewrite, GetRewriteCause()});
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass.cc:397]: I think `RewriteIfX86` is a more meaningful name.
```suggestion
                      CopyAttrsAll, RewriteIfX86, GetRewriteCause()});
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and sorry for the delay! The overall changes look good to me.

cc: @TensorFlow-MKL, @cantonios

File: tensorflow/core/common_runtime/mkl_layout_pass.cc (hunk 1 above)
Comment (renato-arantes) [reply tensorflow/core/common_runtime/mkl_layout_pass.cc]: Thank you for reviewing this PR. I have made the necessary changes and uploaded it.

**Review by cantonios — APPROVED** [review]

**Pull request merged**